### DISCUSSION TITLE
I am learning PHP and having trouble with how this code works

### DISCUSSION THREAD

**Discussion by ghost** [community/community#69941] [Programming Help]
### Body

ow does this work? <?php

//You can even add more Dollar Signs

$Bar = "a";
$Foo = "Bar";
$World = "Foo";
$Hello = "World";
$a = "Hello";

$a; //Returns Hello
$$a; //Returns World
$$$a; //Returns Foo
$$$$a; //Returns Bar
$$$$$a; //Returns a

$$$$$$a; //Returns Hello
$$$$$$$a; //Returns World

//... and so on ...//

?>

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by thevindu-w** [comment]
I assume you referred to [www.php.net/manual/en/language.variables.variable.php](https://www.php.net/manual/en/language.variables.variable.php).
This is called variable variable. Quoting from PHP manual,

> A variable variable takes the value of a variable and treats that as the name of a variable.

According to that,
- The first `$a` returns the value of the variable `a` which is `"hello"`.
- Next, `$$a` returns the value of the variable with the name given by `$a` which is the variable `hello` which is `"world"`.
- Next, `$$$a` returns the value of the variable with the name given by `$$a` which is the variable `world` (as described above) which is `"Foo"`.
- Next, `$$$$a` returns the value of the variable with the name given by `$$$a` which is the variable `Foo` (as described above) which is `"Bar"`.
- Next, `$$$$$a` returns the value of the variable with the name given by `$$$$a` which is the variable `Bar` (as described above) which is `"a"`.

The pattern continues like this.
